### PR TITLE
sh.xaviers.Opentapes

### PR DESCRIPTION
**Pull request by xsaliniere**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

File: sh.xaviers.Opentapes.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    sources:
+      - type: archive
+        # For next release: https://codeberg.org/xaviers/Opentapes/archive/0.x.0.tar.gz
+        url: https://codeberg.org/xaviers/Opentapes/archive/main.tar.gz
+        sha256: 77a5143398b78cc86bc051b1c1372f7a38db8c2da92979375904c47db9ccd2bc
+    buildsystem: simple
+    build-commands:
+      - yarn install
+      - yarn build
+      - cargo install tauri-cli --version 1.0.0-beta.7
```
Comment (hfiguiere) [inline sh.xaviers.Opentapes.yml:40]: this won't work, there is no network access at build time. flatpak-builder-tools should help you create the manifest for the dependencies for both cargo and yarn: https://github.com/flatpak/flatpak-builder-tools

File: sh.xaviers.Opentapes.yml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  - --device=dri
+  - --share=ipc
+  - --socket=wayland
+  - --socket=fallback-x11
+  - --socket=pulseaudio
+  - --share=network
+build-options:
+  append-path: /usr/lib/sdk/node16/bin:/usr/lib/sdk/rust-stable/bin
+  build-args:
+    - --share=network
```
Comment (hfiguiere) [inline sh.xaviers.Opentapes.yml:20]:
```suggestion
```

there is not network at build time.

File: sh.xaviers.Opentapes.desktop [hunk 3]
```diff
@@ -0,0 +1,10 @@
+[Desktop Entry]
+Type=Application
+Name=Opentapes
+Comment=A convergent/adaptive app for streaming Hip-Hop mixtapes. Made with Vue.js & Tauri.
```
Comment (hfiguiere) [inline sh.xaviers.Opentapes.desktop:4]:
```suggestion
Comment=A convergent/adaptive app for streaming Hip-Hop mixtapes.
```

**Review by hfiguiere — COMMENTED** [review]
I'm gonna assume that the appdata and .desktop, and 512x512 icons will be part of the next upstream release.

**Comment by xsaliniere** [comment]
Thanks for the review @hfiguiere I'll do the changes.
I'm not sure why I added the desktop, appdata and the icon in the flathub repo. I probably saw someone else do it on theirs.
I'll remove them since they're useless.

**Comment by xsaliniere** [comment]
Done, it should build successfully without network now.
I've also removed the "Made with Vue & Tauri" sentence from the app description: https://codeberg.org/xaviers/Opentapes/pulls/5.

**Comment by xsaliniere** [comment]
bot, build sh.xaviers.Opentapes

**Comment by xsaliniere** [comment]
bot, build sh.xaviers.Opentapes

**Comment by xsaliniere** [comment]
The x84_64 build works but the aarch64 build doesn't run the app, no output, no window even when adding -v to flatpak run.
Is there any way I can debug what's happening on aarch64 ?

**Comment by xsaliniere** [comment]
I think that I'll have to make it work without the --ignore-scripts yarn option.

**Comment by hfiguiere** [comment]
> The x84_64 build works but the aarch64 build doesn't run the app, no output, no window even when adding -v to flatpak run. Is there any way I can debug what's happening on aarch64 ?

The app starts fine on my RaspberryPi 400.

**Comment by xsaliniere** [comment]
> > The x84_64 build works but the aarch64 build doesn't run the app, no output, no window even when adding -v to flatpak run. Is there any way I can debug what's happening on aarch64 ?
> 
> The app starts fine on my RaspberryPi 400.

Oh well good news then. It must be an issue with my Pinephone installation.

**Comment by xsaliniere** [comment]
So what is the next step?

File: sh.xaviers.Opentapes.yml [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+cleanup:
+  - /node_modules
+  - /dist
+  - /src-tauri/target
+modules:
+  - name: Opentapes
+    sources:
+      - type: archive
+        # For next release: https://codeberg.org/xaviers/Opentapes/archive/0.x.0.tar.gz
+        url: https://codeberg.org/xaviers/Opentapes/archive/main.tar.gz
```
Comment (barthalion) [inline sh.xaviers.Opentapes.yml:32]: Note this URL is not stable and should be changed to point to specific commit instead of branch name.

**Comment by barthalion** [comment]
bot, build sh.xaviers.Opentapes

**Pull request closed without merging**